src/main/java/entities/User.java

```diff
@@ -0,0 +1,170 @@
+        this.isAdmin = admin;
+        User.numUsers = numUsersCreated;
+        User.numUsers ++;
+        this.id = User.numUsers;
+        this.email = email;
+        this.password = password;
+    }
+
+    /*
+    * Returns User's favourites
+    *
+    * @return User's favourites attribute
+    * */
+    public ArrayList getFavourites() {
+        /* Return an ArrayList containing the favourites of this User */
+        return this.favourites;
+    }
+
+    /*
+    * Adds a Post to User's favourites
+    *
+    * @param toAdd The Post to add to User's favourites
+    * */
+    public void addFavourite(Post toAdd) {
+        /* Add Post to this.favourites */
+        this.favourites.add(toAdd);
+    }
+
+    /*
+    * Check if a Post is in User's favourites and remove it if so
+    *
+    * @param toRemove Post to be removed
+    * @return boolean representing success or failure
+    * */
+    public boolean removeFavourite(Post toRemove) {
+        for (int i = 0; i < this.favourites.size(); i++) {
+            if (this.favourites.get(i).equals(toRemove)) {
+                this.favourites.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /*
+    * Return Posts made by User
+    *
+    * @return User's posts attribute
+    * */
+    public ArrayList getPosts() {
+        /* return an ArrayList containing the elements of this.posts */
+        return this.posts;
+    }
+
+    /*
+    * Add a Post to User's posts
+    *
+    * @param toAdd post to be added
+    * */
+    public void addPost(Post toAdd) {
+        /* add toAdd to this.posts */
+        this.posts.add(toAdd);
+    }
+
+    /*
+    * Check if a Post is in User's posts and remove it if so
+    *
+    * @param toRemove post to be removed
+    * @returns a boolean representing success or failure
+    * */
+    public boolean removePost(Post toRemove) {
+        /* Search for toRemove in this.posts and if it is the remove and return true,
+         * otherwise return false */
+        for (int i = 0; i < this.posts.size(); i++) {
+            if (this.posts.get(i).equals(toRemove)) {
+                this.posts.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /*
+    * Checks whether User has admin privledges
+    *
+    * @return a boolean which represents whether User has admin priviledges
+    * */
+    public boolean isAdmin(){
+        return this.isAdmin;
+    }
+
+    /*
+    * Return if User is equal to o
+    *
+    * @param o The Object to be compared to
+    * @return boolean representing whether User is equal to o
+    * */
+    @Override
+    public boolean equals (Object o) {
+        if (this == o) {
+            return true;
+        } else if (o == null) {
+            return false;
+        } else if (!(o instanceof User)) {
+            return false;
+        }  else {
+            return (this.id == ((User) o).id);
```

Comment:
Nit: do the typecasting on another line to make code more readable

Reply:
not sure how to do this so I'll just leave it for now

Reply:
Have something like `User user = (User) o` before line 139 and then just compare this.id and user.id